homeassistant/components/rfxtrx/__init__.py

```diff
@@ -135,21 +134,26 @@ def _ensure_device(value):
 async def async_setup(hass, config):
     """Set up the RFXtrx component."""
     if DOMAIN not in config:
-        hass.data[DATA_RFXTRX_CONFIG] = BASE_SCHEMA({})
         return True
 
-    hass.data[DATA_RFXTRX_CONFIG] = config[DOMAIN]
+    data = {
+        CONF_HOST: config[DOMAIN].get(CONF_HOST),
+        CONF_PORT: config[DOMAIN].get(CONF_PORT),
+        CONF_DEVICE: config[DOMAIN].get(CONF_DEVICE),
+        CONF_DEBUG: config[DOMAIN].get(CONF_DEBUG),
+        CONF_AUTOMATIC_ADD: config[DOMAIN].get(CONF_AUTOMATIC_ADD),
+        CONF_DEVICES: config[DOMAIN][CONF_DEVICES],
+    }
+
+    for device in data[CONF_DEVICES]:
+        if data[CONF_DEVICES][device].get(CONF_OFF_DELAY) is not None:
+            data[CONF_DEVICES][device][CONF_OFF_DELAY] = data[CONF_DEVICES][device][
+                CONF_OFF_DELAY
+            ].total_seconds()
```

Comment:
We could move the conversion to seconds to the config schema instead.

```py
vol.All(cv.timeperiod, cv.positive_timedelta, lambda value: value.total_seconds())
```

Reply:
That's indeed cleaner